### PR TITLE
EOS-13992: ADDB tracepoints for directory operations

### PR DESCRIPTION
**Pull request by sachinpunadikar**
# CORTXFS Change Summary

## Problem Statement
_[EOS-13992](https://jts.seagate.com/browse/EOS-13992):_
_ADDB Tracepoints-Directory Operations_

## Problem Description
_Add ADDB tracepoints for Directory operations_

## Solution Overview
_Directory operations include, directory creation (mkdir), directory removal (rmdir), read the directory contents (readdir) and lookup a specific entry in directory (lookup). ADDB tracepoint added for function entry and points. This provides information on how much time the function / code snippet takes._

### Companion PRs
https://github.com/Seagate/cortx-utils/pull/48
https://github.com/Seagate/cortx-fs-ganesha/pull/40
https://github.com/Seagate/cortx-nsal/pull/23

## Unit Test Cases
_No new test cases added_

## Testing
<details>
  <summary>Click here to see the test execution output</summary>

```
[root@ssc-vm-c-0689 cortx-posix]# ./scripts/test.sh
 --
 --
Configuration Completed
Clean indexes prepared
NSAL Unit tests
NS Tests
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 5
Tests passed = 5
Tests failed = 0

Iterator test
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 2
Tests passed = 2
Tests failed = 0

KVTree test
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 17
Tests passed = 17
Tests failed = 0

CORTXFS Unit tests
Endpoint ops Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.logs
Total tests  = 4
Tests passed = 4
Tests failed = 0

FS Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 3
Tests passed = 3
Tests failed = 0

Directory tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 14
Tests passed = 14
Tests failed = 0

File creation tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 4
Tests passed = 4
Tests failed = 0

Link Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 6
Tests passed = 6
Tests failed = 0

Rename tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 3
Tests passed = 3
Tests failed = 0

Attribute Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 5
Tests passed = 5
Tests failed = 0

Xattr file Tests
Test results are logged to /var/log/cortx/test/ut/xattr_file_ops.log
Total tests  = 9
Tests passed = 9
Tests failed = 0

Xattr dir Tests
Test results are logged to /var/log/cortx/test/ut/xattr_dir_ops.log
Total tests  = 9
Tests passed = 9
Tests failed = 0

IO tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 7
Tests passed = 7
Tests failed = 0

DSAL Unit tests
Dsal basic test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 11
Tests passed = 11
Tests failed = 0

Dsal IO test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 1
Tests passed = 1
Tests failed = 0

Dsal space stats test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 2
Tests passed = 2
Tests failed = 0


```

</details>

## Checklist
- [x] **Compilation:** _This patch does not break compilation_
- [x] **Merge conflicts:** _None_
- [ ] **Code review:** _In progress_
- [X] **Sanity Testing:** _Done as mentioned in testing above_
- [X] **Documentation:** _None_

### CODE REVIEW

File: src/cortxfs/cortxfs_fh.c [hunk 1]
```diff
@@ -215,6 +217,18 @@ int cfs_fh_lookup(const cfs_cred_t *cred, struct cfs_fh *parent_fh,
 	return rc;
```
Comment (pratyush-seagate) [inline src/cortxfs/cortxfs_fh.c:217]: Should we trace rc?

File: src/cortxfs/cortxfs_ops.c [hunk 2]
```diff
@@ -281,44 +279,70 @@ int cfs_readdir(struct cfs_fs *cfs_fs,
 	return rc;
```
Comment (pratyush-seagate) [inline src/cortxfs/cortxfs_ops.c:279]: Should we trace rc?

File: src/cortxfs/cortxfs_fh.c [hunk 3]
```diff
@@ -159,8 +160,9 @@ int cfs_fh_from_ino(struct cfs_fs *fs, const cfs_ino_t *ino_num,
...
 }
 
-int cfs_fh_lookup(const cfs_cred_t *cred, struct cfs_fh *parent_fh,
-                  const char *name, struct cfs_fh **fh)
+static inline int __cfs_fh_lookup(const cfs_cred_t *cred,
+		struct cfs_fh *parent_fh,
```
Comment (vimalraghwani20) [inline src/cortxfs/cortxfs_fh.c:164]: check indentation

**Review by vimalraghwani20 — APPROVED** [review]

File: src/cortxfs/cortxfs_fh.c (hunk 1 above)
Comment (sachinpunadikar) [reply src/cortxfs/cortxfs_fh.c:217]: not required at this stage.

File: src/cortxfs/cortxfs_ops.c (hunk 2 above)
Comment (sachinpunadikar) [reply src/cortxfs/cortxfs_ops.c:279]: not required at this stage.

File: src/cortxfs/cortxfs_fh.c (hunk 3 above)
Comment (sachinpunadikar) [reply src/cortxfs/cortxfs_fh.c:164]: will address this

**Review by vimalraghwani20 — APPROVED** [review]
looks good to me

**Review by pratyush-seagate — APPROVED** [review]
LGTM, +1

**Review by s-arya — APPROVED** [review]

**Pull request merged**